### PR TITLE
joyent/node-krill#5: add support for getting a map of field names to values

### PR DESCRIPTION
**Pull request by misterdjules**
This is a work in progress. If that looks good, then I imagine we'll want to bump at least the minor version number in `package.json` at some point

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-	    "type": "git",
-	    "url": "http://github.com/joyent/node-krill.git"
-	},
-	"main": "lib/krill.js",
-	"dependencies": {
-	    "extsprintf": "1.0.2",
-	    "jsprim": "0.5.1",
-	    "verror": "1.3.6"
-	}
+  "name": "krill",
```
Comment (davepacheco) [inline package.json]: Same comment on the reformatting here.

**Comment by davepacheco** [comment]
I think this is okay, but I don't understand what it's useful for.  Can you explain how it would be used?

**Comment by misterdjules** [comment]
@davepacheco I tried to describe my use case in [the corresponding issue](https://github.com/joyent/node-krill/issues/5):

> [It] allow[s] me to iterate through this map, get the validation function corresponding to each field name, and call that validation function for each value used for that field in the input predicate.

Does that clarify the use case I have for it?

**Comment by misterdjules** [comment]
@davepacheco Added a new separate commit according to your code review comments.

**Comment by davepacheco** [comment]
Looks good!  Thanks.  Assuming it's `make prepush` clean, feel free to integrate and publish.  (I can also take care of that.)

**Comment by misterdjules** [comment]
Merged with 7c5efeaa13ec5dcdd63aaf198bf6d8623ce27a4d, thank you!

**Pull request closed without merging**